device/core/test/_module_client_test.js

```diff
@@ -5,12 +5,43 @@
 
 var EventEmitter = require('events');
 var assert = require('chai').assert;
-var errors = require('azure-iot-common').errors;
 var sinon = require('sinon');
+var FakeTransport = require('./fake_transport.js');
+var Message = require('azure-iot-common').Message;
+var ModuleClient = require('../lib/module_client').ModuleClient;
+var errors = require('azure-iot-common').errors;
 var ModuleClient = require('../lib/module_client').ModuleClient;
 var IotEdgeAuthenticationProvider = require('../lib/iotedge_authentication_provider').IotEdgeAuthenticationProvider;
 
-describe('ModuleClient', function() {
+describe('ModuleClient', function () {
+  var sharedKeyConnectionString = 'HostName=host;DeviceId=id;ModuleId=modId;SharedAccessKey=key';
+  var sharedAccessSignature = '"SharedAccessSignature sr=hubName.azure-devices.net/devices/deviceId/modules/moduleId&sig=s1gn4tur3&se=1454204843"';
+
+  describe('#fromConnectionString', function () {
+    /*Tests_SRS_NODE_INTERNAL_CLIENT_05_006: [The fromConnectionString method shall return a new instance of the InternalClient object, as by a call to new InternalClient(new Transport(...)).]*/
+    it('returns an instance of ModuleClient', function () {
+      var client = ModuleClient.fromConnectionString(sharedKeyConnectionString, FakeTransport);
+      assert.instanceOf(client, ModuleClient);
+    });
+  });
+
+
+  describe('#fromSharedAccessSignature', function () {
+    /*Tests_SRS_NODE_INTERNAL_CLIENT_16_030: [The fromSharedAccessSignature method shall return a new instance of the InternalClient object] */
+    it('returns an instance of ModuleClient', function () {
+      var client = ModuleClient.fromSharedAccessSignature(sharedAccessSignature, FakeTransport);
+      assert.instanceOf(client, ModuleClient);
+    });
+  });
+
+  describe('#fromAuthenticationProvider', function () {
+    /*Tests_SRS_NODE_INTERNAL_CLIENT_16_091: [The `fromAuthenticationProvider` method shall return a `InternalClient` object configured with a new instance of a transport created using the `transportCtor` argument.]*/
```

Comment:
>transport created using the `transportCtor` argument [](start = 161, length = 52)

For completeness, do you want to assert that it has a transport created using the transportCtor?